rotkehlchen/data_migrations/utils.py

```diff
@@ -31,15 +36,21 @@ def update_data_and_detect_accounts(
     progress_handler.new_step('Fetching new spam assets and rpc data info')
     rotki.data_updater.check_for_updates(updates=[UpdateType.SPAM_ASSETS, UpdateType.RPC_NODES])
 
-    with rotki.data.db.conn.write_ctx() as write_cursor:
-        rotki.data.db.add_external_service_credentials(
-            write_cursor=write_cursor,  # add temporary etherscan key for the given chains
-            credentials=external_service_credentials,
+    with rotki.data.db.conn.read_ctx() as cursor:
+        cursor.execute(
+            'SELECT COUNT(*) FROM external_service_credentials WHERE name=?',
+            (ExternalService.ETHERSCAN.name.lower(),),
         )
+        if (needs_etherscan_key := cursor.fetchone()[0] == 0):
+            with rotki.data.db.conn.write_ctx() as write_cursor:
+                rotki.data.db.add_external_service_credentials(
+                    write_cursor=write_cursor,  # add temporary etherscan key for the given chains
+                    credentials=[ExternalServiceApiCredentials(
+                        service=ExternalService.ETHERSCAN,
+                        api_key=ApiKey(ROTKI_PACKAGED_KEY),
+                    )],
+                )
 
     rotki.chains_aggregator.detect_evm_accounts(progress_handler, chains=chains)
-    external_services = []
-    for external_service, _, _ in external_service_credentials:
-        external_services.append(external_service)
-
-    rotki.data.db.delete_external_service_credentials(external_services)
+    if needs_etherscan_key:
```

Comment:
shouldn't this be `not needs_etherscan_key`? since you're trying to delete the temporary key added earlier